js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.js

```diff
@@ -39,25 +35,27 @@ const MinimumOrderCard = ( props ) => {
 			( shippingRate ) => shippingRate.country
 		);
 
-		/**
-		 * If group length is 1, we render the group,
-		 * regardless of threshold is defined or not.
-		 */
+		// If group length is 1, we render the group,
+		// regardless of threshold is defined or not.
 		if ( groups.length === 1 ) {
 			return (
 				<MinimumOrderInputControl
 					countryOptions={ countryOptions }
 					value={ groups[ 0 ] }
-					onChange={ getMinimumOrderChangeHandler(
-						value,
-						onChange,
-						groups[ 0 ]
-					) }
-					onDelete={ getMinimumOrderDeleteHandler(
-						value,
-						onChange,
-						groups[ 0 ]
-					) }
+					onChange={ ( newGroup ) =>
+						onChange(
+							changeMinimumOrderGroup(
+								value,
+								groups[ 0 ],
+								newGroup
+							)
+						)
+					}
+					onDelete={ () =>
+						onChange(
+							changeMinimumOrderGroup( value, groups[ 0 ] )
+						)
+					}
```

Comment:
💅 This comment applies to the add, edit, and delete operations (`onSubmit`, `onChange`, and `onDelete`) in this `MinimumOrderCard` component.

Instead of using inline arrow functions (and having a bit of duplicated code across the two `onChange` and `onDelete`), I would still prefer to have a dedicated "handlers" code block so that it is easier to see how those add/edit/delete operations is mapped to the `changeMinimumOrderGroup` call. In case those operations need to be modified, they just need to be modified in their corresponding handler code instead of the duplicated inline arrow function.

Refer to the screenshot below:

<img width="740" alt="image" src="https://user-images.githubusercontent.com/417342/168635459-e62d0ed1-2429-4264-b6b5-171c9cee1e24.png">

Here's the code if you want to copy and paste: 

```js
		const getAddHandler = ( newGroup ) => {
			const newValue = changeMinimumOrderGroup( value, null, newGroup );
			onChange( newValue );
		};

		const getChangeHandler = ( oldGroup ) => ( newGroup ) => {
			const newValue = changeMinimumOrderGroup(
				value,
				oldGroup,
				newGroup
			);
			onChange( newValue );
		};

		const getDeleteHandler = ( oldGroup ) => () => {
			const newValue = changeMinimumOrderGroup( value, oldGroup );
			onChange( newValue );
		};
```

So with the above code, we can easily see how add/change/delete operations is mapped to `changeMinimumOrderGroup`, and how they depend on `oldGroup` and `newGroup`. There are only three `changeMinimumOrderGroup` calls. With the inline arrow functions, IMO it is not as clear, and there are five `changeMinimumOrderGroup` calls.

Reply:
> Instead of using inline arrow functions (and having a bit of duplicated code across the two `onChange` and `onDelete`), I would still prefer to have a dedicated "handlers" code block so that it is easier to see how those add/edit/delete operations is mapped to the `changeMinimumOrderGroup` call. In case those operations need to be modified, they just need to be modified in their corresponding handler code instead of the duplicated inline arrow function.

I'd address duplication as mentioned in https://github.com/woocommerce/google-listings-and-ads/pull/1499#issuecomment-1129049643 by removing the duplication in the rendering case when those handlers are called. Then we will end up with 3 different use cases of `changeMinimumOrderGroup` and 3 calls - no duplication whatsoever.

Then, changing:
```jsx
onDelete={ () =>
	  onChange(
		  changeMinimumOrderGroup( value, groups[ 0 ] )
	  )
}
```
to 
```jsx
const getDeleteHandler = ( oldGroup ) => () => {
	const newValue = changeMinimumOrderGroup( value, oldGroup );
	onChange( newValue );
};
// ~ 40 lines of unrelated code
onDelete={ getDeleteHandler( group ) }
```
Does not bring many benefits, as 
- I need to "jump around" to check what `onDelete` actually does, the code is more dislocated and less explicit
- the amount of lines is exactly the same


The one reason that makes me a bit more convinced about that is because we had it that way before #1410, so making it as it was would minimize the changes in this PR(s). Then we can address that together with removing the duplication of renderings.

Reply:
> I'd address duplication as mentioned in [#1499 (comment)](https://github.com/woocommerce/google-listings-and-ads/pull/1499#issuecomment-1129049643) by removing the duplication in the rendering case when those handlers are called. Then we will end up with 3 different use cases of `changeMinimumOrderGroup` and 3 calls - no duplication whatsoever.

I'm not sure how you would like to address the duplication. I'm guessing, maybe you are considering to do some mutations on the `groups` array, so that there is only one `return` in the `renderGroups` function? 🤔 That was the approach we previously had for shipping rates groups (not minimum order groups).

If that is the idea, I actually intentionally want to avoid that, and it is because the `if ( groups.length === 1 )` is a _special case_ where we want to show the `MinimumOrderInputControl` even when there is no MinimumOrderGroup with defined threshold, and we want to return early for such special case. With this, `groups` array is not mutated, and its meaning is always the same, as in "`groups` is the result from calling `groupShippingRatesByMethodFreeShippingThreshold` function". If we were to mutate the `groups` array and use one `return` statement, then the `groups` array meaning would change and wouldn't be as straightforward, it makes it a little harder to follow and understand, and the one `return` statement wouldn't clearly show we have a special rendering for the special case.

Reply:
> I'm not sure how you would like to address the duplication. I'm guessing, maybe you are considering to do some mutations on the groups array, so that there is only one return in the renderGroups function? thinking That was the approach we previously had for shipping rates groups (not minimum order groups).

That's something to explore and discuss in a separate PR :)

Reply:
I defined dedicated handlers in https://github.com/woocommerce/google-listings-and-ads/pull/1499/commits/8c31c29840314c02b282212c0711b6f7e0e02193